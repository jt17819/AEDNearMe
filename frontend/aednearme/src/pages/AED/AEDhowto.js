import { Grid, Container, makeStyles, Paper } from '@material-ui/core';

import DefibVid from '../../components/DefibVid/DefibVid'

import React, { Fragment } from 'react'

const useStyles = makeStyles(() => ({

  title: {
    paddingBottom: '5px',
  },

  text: {
    fontSize: '16px'
  },

  btn_box:{
    marginTop: "30px",
    display: 'flex',
    justifyContent: 'space-between',
  },

  primary: {
    border: "none",
    background: "#273746",
    color: '#fff',
    borderRadius: "0px",
    padding: "10px 20px",
    "&:hover": {
      background: "#34495E",
    }
  },
  
  secondary: {
    border: "none",
    background: "aliceblue",
    color: "#ba181b",
    borderRadius: "0px",
    padding: "10px 20px",
    "&:hover": {
      background: "aliceblue",
    }
  },

}));

const AEDhowto = () => {

  const classes = useStyles();

  return (
    <Fragment>

      <Container className="pb-5" style={{
        paddingBottom: "100px",
        paddingTop: "20px"
      }}>
      
      <h1 className={classes.title} >How to use an AED</h1>


          <Grid container spacing={4}>

            <Grid item md={5} >
            
            <h3 className={classes.text}>
            Step 1: Have someone continue to perform CPR and follow the instructions of the defibrillator. 
            </h3>

            <h3 className={classes.text}>
            Step 2: Remove clothing from person's chest and find the pads. Peel off the plastic and place the pads indicated by the pictures onto the person. (One pad of the right side under collarbone, second pad on the left side below armpit)
            </h3>

            <h3 className={classes.text}>
            Step 3: AED will analyse the person's heart rhythm. Stop chest compressions and wait for the AED's instructions
            </h3>

            <h3 className={classes.text}>
            Step 4: If shock is advised, tell everyone to stand back and clear of the person. After shock is delivered start chest compressions when told to do so by AED. 
            </h3>

            <h3 className={classes.text}>
            Step 5: Leave the AED on as it will reanalyse heart rhythm and follow the instructions until help arrives
            </h3>

            <h3>Read more here at <a id="choking-link" href="https://www.sja.org.uk/get-advice/first-aid-advice/how-to/how-to-use-a-defibrillator/?category=12349" target="_blank">St. John's Ambulance</a>
            </h3>


            </Grid>

            <Grid
              item
              md={7}
              xs={12}
              className=""
              > 
              <div>
                <DefibVid />
              </div>  
            </Grid>

            
          </Grid>
        </Container>


    </Fragment>
      
  )
}

export default AEDhowto