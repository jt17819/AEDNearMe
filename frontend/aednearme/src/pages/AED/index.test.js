import { render, fireEvent, screen, cleanup, within } from '@testing-library/react';
import { MemoryRouter, Router as Router } from 'react-router-dom';
import ReactDOM from 'react-dom';
import { createMemoryHistory } from 'history'
import {default as AEDhowto} from './AEDhowto';
import axios from 'axios'


describe('AEDhowto', () => {

  // beforeEach(() => {
  //   render(<App />, { wrapper: MemoryRouter })
  // });


  // Rendering

  it('renders welcome message', () => {
    render(<AEDhowto />, { wrapper: MemoryRouter })
    expect(screen.getByText('How to use an AED')).toBeInTheDocument();
  });

//   test('list contains 1 buttons', () => {
//     render(<AEDhowto />, { wrapper: MemoryRouter });
  
//     // const listElement = screen.getAllByRole('button');
//     const listItems = screen.getAllByRole('button');
  
    
//     expect(listItems.length).toEqual(1);
//   });



  // Snapshot

  it('should take a snapshot', () => {
    render(<AEDhowto />, { wrapper: MemoryRouter })
    const { asFragment } = render(<AEDhowto />, { wrapper: MemoryRouter })
    
    expect(asFragment(<AEDhowto />)).toMatchSnapshot()
  })

  

  
})

