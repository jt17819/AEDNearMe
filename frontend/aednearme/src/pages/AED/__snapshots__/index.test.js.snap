// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AEDhowto should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root pb-5 MuiContainer-maxWidthLg"
    style="padding-bottom: 100px; padding-top: 20px;"
  >
    <h1
      class="makeStyles-title-6"
    >
      How to use an AED
    </h1>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-5"
      >
        <h3
          class="makeStyles-text-7"
        >
          Step 1: Have someone continue to perform CPR and follow the instructions of the defibrillator.
        </h3>
        <h3
          class="makeStyles-text-7"
        >
          Step 2: Remove clothing from person's chest and find the pads. Peel off the plastic and place the pads indicated by the pictures onto the person. (One pad of the right side under collarbone, second pad on the left side below armpit)
        </h3>
        <h3
          class="makeStyles-text-7"
        >
          Step 3: AED will analyse the person's heart rhythm. Stop chest compressions and wait for the AED's instructions
        </h3>
        <h3
          class="makeStyles-text-7"
        >
          Step 4: If shock is advised, tell everyone to stand back and clear of the person. After shock is delivered start chest compressions when told to do so by AED.
        </h3>
        <h3
          class="makeStyles-text-7"
        >
          Step 5: Leave the AED on as it will reanalyse heart rhythm and follow the instructions until help arrives
        </h3>
        <h3>
          Read more here at 
          <a
            href="https://www.sja.org.uk/get-advice/first-aid-advice/how-to/how-to-use-a-defibrillator/?category=12349"
            id="choking-link"
            target="_blank"
          >
            St. John's Ambulance
          </a>
        </h3>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7"
      >
        <div>
          <div>
            <div
              style="width: 100%; height: 360px;"
            >
              <div
                style="width: 100%; height: 100%;"
              >
                <div />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
